### DISCUSSION TITLE
E3SM single column model sequential runs

### DISCUSSION THREAD

**Discussion by zhang25** [E3SM-Project/E3SM#7903] [E3SM model help]
Hi there, 

  We need to run thousands of E3SM single column model (SCM) sequentially as the new SCM run's parameter values need to be updated according to the results of the previous SCM run.  We allocate one node in the queue for the whole workflow.  However, using "case.submit" for each of SCM runs, it will wait in the queue and allocate another new node to finish the calculation. At the mean time,  the workflow is waiting on the original node for the new SCM run to finish because this workflow has to be done sequentially.  This waiting and new allocation of nodes significantly slows down the whole workflow and most of time it will break the pipe of the workflow since it is waiting too long.  

we are asking is there any way that we can keep all the sequential SCM calculations at this original one node that the workflow allocated already?
or in other words, we do not have to call for another new node for every SCM run, but all the SCM runs will be performed at the same node of the workflow.       

  This is an important setup E3SM should consider, as SCM is an important tool for many users.  Especially running many SCM on one single node continuously will be a great use of node allocations in a queue. This is important for the machine learning parametric tuning purpose, e.g., using MCMC sampling.  SCM has a great advantage, e.g., it only takes 30 seconds to finish one diurnal case simulation. 

  If there is any thought how this could be done, please do not hesitate to share with us.  Many thanks!

**Comment by golaz** [comment]
Hi @zhang25,

Yes, you can do that. Instead of using `case.submit` to start your SCM simulations, you should use `case.submit --no-batch`. The `--no-batch` option will cause E3SM to run in your existing job (slurm allocation), rather than be submitted as a separate job.

We did something similar for the E3SMv2 large-ensemble (but were running jobs in parallel rather than sequentially).

For more information:
https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3255271449/v2.LR.historical+LE

and the actual run script:
https://github.com/E3SM-Project/SimulationScripts/blob/master/archive/v2/v2.LR.LE_bundle.sh

**Comment by mikaelwitte** [comment]
Hi @zhang25,
Did you get this method to work? @mchinita and I are trying to start many SCM jobs in parallel on a single node and haven't been successful yet.

**Comment by zhang25** [comment]
@mikaelwitte Yes, we run multiple SCM jobs on one node, that's doable, however it is very slow and we have not figure it out why it is like that yet.

**Comment by tangq** [comment]
@zhang25, is it possible that the slowness is due to the memory limit? One way to check it is to use the interactive node.
